diffus-derive/src/lib.rs

```diff
@@ -64,109 +62,129 @@ fn renamed_field_ident(
     }
 }
 
-fn renamed_field_idents(
-    fields: &syn::Fields,
-    prefix: &str,
-) -> Output {
-    let field_idents = fields.iter().enumerate()
+fn renamed_field_idents(fields: &syn::Fields, prefix: &str) -> Output {
+    let field_idents = fields
+        .iter()
+        .enumerate()
         .map(|enumerated_field| renamed_field_ident(enumerated_field, prefix));
 
     quote! { #(#field_idents),* }
 }
 
-fn matches_all_copy(
-    fields: &syn::Fields
-) -> Output {
-    let edit_fields_copy = fields.iter().enumerate()
-        .map(|enumerated_field| {
-            let field_ident = field_ident(enumerated_field, "");
+fn matches_all_copy(fields: &syn::Fields) -> Output {
+    let edit_fields_copy = fields.iter().enumerate().map(|enumerated_field| {
+        let field_ident = field_ident(enumerated_field, "");
 
-            quote! { #field_ident @ diffus::edit::Edit::Copy }
-        });
+        quote! { #field_ident @ diffus::edit::Edit::Copy }
+    });
 
     quote! {
         ( #(#edit_fields_copy),* ) => diffus::edit::Edit::Copy
     }
 }
 
-fn field_diffs(
-    fields: &syn::Fields
-) -> Output {
-    let field_diffs = fields.iter()
-        .enumerate()
-        .map(|(index, field)| {
-            let field_name = match field {
-                syn::Field { ident: Some(ident), .. } => quote! { #ident },
-                syn::Field { ident: None, .. } => {
-                    let ident = unnamed_field_name(index);
-
-                    quote! { #ident }
-                }
-            };
+fn field_diffs(fields: &syn::Fields) -> Output {
+    let field_diffs = fields.iter().enumerate().map(|(index, field)| {
+        let field_name = match field {
+            syn::Field {
+                ident: Some(ident), ..
+            } => quote! { #ident },
+            syn::Field { ident: None, .. } => {
+                let ident = unnamed_field_name(index);
 
-            quote! {
-                self.#field_name.diff(&other.#field_name)
+                quote! { #ident }
             }
-        });
+        };
+
+        quote! {
+            self.#field_name.diff(&other.#field_name)
+        }
+    });
 
     quote! { #(#field_diffs),* }
 }
 
-
-fn unnamed_field_ident(
-    i: usize,
-) -> syn::Ident {
+fn unnamed_field_ident(i: usize) -> syn::Ident {
     format_ident!("x{}", i as u32)
 }
-fn unnamed_field_name(
-    i: usize,
-) -> syn::Lit {
+fn unnamed_field_name(i: usize) -> syn::Lit {
     syn::parse_str(&format!("{}", i as u32)).unwrap()
 }
 
+fn input_lifetime(generics: &syn::Generics) -> Option<syn::Lifetime> {
+    let mut lifetimes = generics.params.iter().filter_map(|generic_param| {
+        if let syn::GenericParam::Lifetime(syn::LifetimeDef { lifetime, .. }) = generic_param {
+            Some(lifetime)
+        } else {
+            None
+        }
+    });
+
+    // FIXME rm cloned
```

Comment:
Should be pretty easy, just change above signature:
```rust
fn input_lifetime(generics: &syn::Generics) -> Option<&syn::Lifetime> {
```

Since you get lifetime elision